### ISSUE TITLE
Type generation (e.g. /types/csharp) fails invisibly with some data types

### ISSUE THREAD

**Issue by jsobell** [ServiceStack/Issues#338]
The generator for types scans for 'malicious' characters in the typename, including a space.  In the case of collections such as Dictionary<string, string> this causes an internal server exception when accessing the types page, or when adding the service through the VS 'Add ServiceStackReference...' menu.
Creating a substitute class to encapsulate the type is a workaround.

Example error generated:
```
[TypesCSharp: 24/08/2015 5:15:01 AM]: [REQUEST: {BaseUrl:"http://localhost:8110"}]
System.IO.InvalidDataException: MetaData is potentially malicious. 
Expected token, Received: Dictionary<string, string> at
ServiceStack.NativeTypes.MetadataExtensions.SafeToken(String token) at
ServiceStack.NativeTypes.CSharp.CSharpGenerator.NameOnly(String type, Boolean includeNested) at
ServiceStack.NativeTypes.CSharp.CSharpGenerator.TypeAlias(String type, Boolean includeNested) at
ServiceStack.NativeTypes.CSharp.CSharpGenerator.Type(String type, String[] genericArgs, Boolean includeNested) at
ServiceStack.NativeTypes.CSharp.CSharpGenerator.<>c__DisplayClass1e.<GetCode>b__f() at
ServiceStack.NativeTypes.CSharp.CSharpGenerator.AppendType(StringBuilderWrapper& sb, MetadataType type, String lastNS, List`1 allTypes, CreateTypeOptions options) at
ServiceStack.NativeTypes.CSharp.CSharpGenerator.GetCode(MetadataTypes metadata, IRequest request) at ServiceStack.NativeTypes.NativeTypesService.Any(TypesCSharp request) at
lambda_method(Closure , Object , Object ) at ServiceStack.Host.ServiceRunner`1.Execute(IRequest request, Object instance, TRequest requestDto)
```

**Comment by mythz** [comment]
that's odd, what version of ServiceStack are you using? if this is one of your DTO's could you also post the source code?

**Comment by jsobell** [comment]
Using 4.0.42.0, but it only occurs if you're returning a Dictionary as the top level object of a response.
e.g. (modified from your Hello example):
```
namespace SampleProblem.ServiceModel
{
    [Route("/hello/{Name}")]
    public class Hello : IReturn<Dictionary<string, string>>
    {
        public string Name { get; set; }
    }

    public class MyServices : Service
    {
        public Dictionary<string,string> Any(Hello request)
        {
            var test = new Dictionary<string, string> {{"Hi", "Hello, {0}!".Fmt(request.Name)}};
            return test;
        }
    }
}
```

If the return type is complex like this the exception is raised internally, but you don't get to see it unless you enable DebugMode to see the internal error :)

**Comment by mythz** [comment]
Cheers, should now be resolved with this commit: https://github.com/ServiceStack/ServiceStack/commit/8eaeaa84857d35619fccc0ff89f8565dd809cb6c

This change is available from v4.0.43+ that's now [available on MyGet](https://github.com/ServiceStack/ServiceStack/wiki/MyGet)
